sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala

```diff
@@ -324,6 +327,32 @@ class RapidsExecutorPlugin extends ExecutorPlugin with Logging {
     }
   }
 
+  // Try to run nvidia-smi when task fails due to a cuda exception.
+  private def logGpuDebugInfo() = {
+    try {
+      val nvidiaSmiStdout = new StringBuilder
+      val nvidiaSmiStderr = new StringBuilder
+      val cmd = "nvidia-smi".run(
+        ProcessLogger(s => nvidiaSmiStdout.append(s + "\n"), s => nvidiaSmiStderr.append(s + "\n")))
+      val cmdFuture = concurrent.Future(concurrent.blocking(cmd.exitValue()))
```

Comment:
It seems weird that this needs to use a thread pool. Should it not invoke the command once per process while blocking the task thread?  

Also, I believe we should `synchronize` this function wholesale, and `System.exit(20)` inside of the synchronized block. That way, any other task thread won't try to print `nvidia-smi`, instead they will all arrive here, block, then print nvidia-smi once and then exit the process. Thoughts?

Reply:
If we do move the `System.exit` into this function, we could change it into `logGpuDebugInfoAndExit`

Reply:
@jbrennan333 pointed out the use of the thread pool here is to address the timeout concern (https://github.com/NVIDIA/spark-rapids/pull/6606#discussion_r978892815) from @gerashegalov. That seems fine to me, I would probably add a comment to make it clear this is done for that reason.

Reply:
I updated this to just do the wait in a while loop (using Process.isAlive).